File: 724. Find Pivot Index/Approach_1.java
class Solution {
    public int pivotIndex(int[] nums) {
        int n = nums.length;
        if (n == 1) {
            return 0;
        }

        int prefix[] = new int[n];
        int suffix[] = new int[n];

        for (int i = 0; i < n; i++) {
            if (i == 0) {
                prefix[i] = 0;
            } else if (i == 1) {
                prefix[i] = nums[i-1];
            } else {
                prefix[i] = nums[i - 1] + prefix[i - 1];
            }
        }

        for (int i = n - 1; i >= 0; i--) {
            if (i == n - 1) {
                suffix[i] = 0;
            } else if (i == n - 2) {
                suffix[i] = nums[i + 1];
            } else {
                suffix[i] = nums[i + 1] + suffix[i + 1];
            }
        }

        for (int i = 0; i < n; i++) {
            if (prefix[i] == suffix[i]) {
                System.out.println(prefix[i]);
                return i;
            }
        }
        return -1;
    }
}

File: 4031. Find All Numbers Disappeared in an Array II/Approach_1.cpp
class Solution {
public:
    vector<vector<int>> findDisappearedNumbers(vector<int>& nums, int lower, int upper) {
        vector<vector<int>> ans;
      
        int fixsize = 1e5;
        vector<int>hash(fixsize+1,0);

        for(int i = 0 ; i < nums.size(); i++){
            hash[nums[i]] = 1;
        }

        int start = lower;
        while(start <= upper){
            int end = start;
            while(end <= upper && hash[end] == 0) end++;
            if(start != end) {
                ans.push_back({start,end-1});
            }
            start = end + 1;
        }
        return ans;
    }
};